agcaballero/portfolio/src/main/webapp/script.js

```diff
@@ -11,3 +11,46 @@
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
+
+/* Returns the link to a random post on the blog */
+function randomPost() {
+  // array with all post links
+  const posts = ['#post1-lnk', '#post2-lnk', '#post3-lnk', '#post4-lnk'];
+
+  // returns a random link
+  return posts[Math.floor(Math.random() * posts.length)];
+}
+
+/* Tests randomPost() by verifying it spits out only one of the random post
+options & logging the percentage of times it returns each */
+function testRandomPost(trials = 10000) {
+  const LENGTH = 4;
```

Comment:
I like how you tried to follow constant naming convention, but my feeling is that this doesn't count as a constant, since it's used only locally within a single function.

I can infer that LENGTH is the number of items in `posts` in the randomPost() function, but a more readable strategy, and one requiring less maintenance, would be to declare the posts separately, either as a constant or the return value of another function:

```
function postLinks() {
  return ['#post1-lnk', '#post2-lnk', '#post3-lnk', '#post4-lnk'];
}

function randomPost() {
  // array with all post links
  const posts = postLinks();

  // returns a random link
  return posts[Math.floor(Math.random() * posts.length)];
}

function testRandomPost(trials = 10000) {
  const postLinks = postLinks();
  const postLinkCount = postLinks.length;
  ...
}
```

This makes it clearer what `4` represents, and also avoid the need to update this value, if you add more post links.

Reference:
https://google.github.io/styleguide/jsguide.html#naming-constant-names

Reply:
Done